internal/app/subsystems/aio/store/sqlite/sqlite.go

```diff
@@ -215,6 +227,41 @@ const (
 
 	NOTIFICATION_DELETE_STATEMENT = `
 	DELETE FROM notifications WHERE id = ? AND promise_id = ?`
+
+	SCHEDULE_INSERT_STATEMENT = `
+	INSERT INTO schedules 
+		(id, desc, cron, last_run_time, next_run_time, created_on)
+	VALUES 
+		(?, ?, ?, ?, ?, ?)
+	ON CONFLICT(id) DO NOTHING`
+
+	SCHEDULE_UPDATE_STATEMENT = `
+	UPDATE
+		schedules
+	SET
+		desc = ?, cron = ?, last_run_time = ?, next_run_time = ?
+	WHERE
+		id = ? AND (last_run_time IS NULL OR last_run_time < ?)`
```

Comment:
What if we simplify the condition to `WHERE id = ? AND last_run_time = ?`. We should know this value and if it is any other value that means someone else beat us to it